python/lsst/ctrl/execute/slurmPlugin.py

```diff
@@ -432,72 +359,84 @@ def smallGlideinsFromJobPressure(self, generatedSlurmFile):
 
-        except Exception as exc:
-            raise type(exc)("Problem querying condor schedd for jobs") from None
+            print(f"smallGlideins: The final TotalCores is {totalCores}")
 
-        print(f"smallGlideins: The final TotalCores is {totalCores}")
+            # The number of Glideins needed to service the detected Idle jobs
+            # is "numberOfGlideins"
+            numberOfGlideins = math.ceil(totalCores / autoCPUs)
+            print(f"smallGlideins: Number for detected jobs is {numberOfGlideins}")
 
-        # The number of Glideins needed to service the detected Idle jobs
-        # is "numberOfGlideins"
-        numberOfGlideins = math.ceil(totalCores / autoCPUs)
-        print(f"smallGlideins: Number for detected jobs is {numberOfGlideins}")
+            jobname = f"glide_{auser}"
 
-        # Check Slurm queue Running glideins
-        jobname = f"glide_{auser}"
-        existingGlideinsRunning = 0
-        batcmd = f"squeue --noheader --states=R  --name={jobname} | wc -l"
-        print("The squeue command is: %s " % batcmd)
-        try:
-            resultR = subprocess.check_output(batcmd, shell=True)
-        except subprocess.CalledProcessError as e:
-            print(e.output)
-        existingGlideinsRunning = int(resultR.decode("UTF-8"))
+            # Check Slurm queue Running glideins
+            existingGlideinsRunning = self.countRunningSlurmJobs(jobname)
+
+            # Check Slurm queue Idle Glideins
+            existingGlideinsIdle = self.countIdleSlurmJobs(jobname)
 
-        # Check Slurm queue Idle Glideins
-        existingGlideinsIdle = 0
+            print(f"smallGlideins: existingGlideinsRunning {existingGlideinsRunning}")
+            print(f"smallGlideins: existingGlideinsIdle {existingGlideinsIdle}")
+
+            # The number of Glideins needed to service the detected
+            # Idle jobs is "numberOfGlideins" less the existing Idle glideins
+            numberOfGlideinsReduced = numberOfGlideins - existingGlideinsIdle
+            print(f"smallGlideins: Target Number to submit {numberOfGlideinsReduced}")
+
+            # The maximum number of Glideins that we can submit with
+            # the imposed threshold (maxNumberOfGlideins)
+            # is maxSubmitGlideins
+            existingGlideins = existingGlideinsRunning + existingGlideinsIdle
+            maxSubmitGlideins = maxNumberOfGlideins - existingGlideins
+            print(f"smallGlideins: maxNumberOfGlideins {maxNumberOfGlideins}")
+            print(f"smallGlideins: maxSubmitGlideins {maxSubmitGlideins}")
+
+            # Reduce the number of Glideins to submit if threshold exceeded
+            if numberOfGlideinsReduced > maxSubmitGlideins:
+                numberOfGlideinsReduced = maxSubmitGlideins
+                print("smallGlideins: Reducing due to threshold.")
+            print(
+                f"smallGlideins: Number of Glideins to submit is {numberOfGlideinsReduced}"
+            )
```

Comment:
I may be wrong, but Judging by earlier code, it feels like all these messages (lines 377 - 399) should be printed out only if verbose is True.